alea/models/blueice_extended_model.py

```diff
@@ -431,6 +432,23 @@ def _set_efficiency(self, source: dict, ll):
             )
         ll.add_shape_parameter(efficiency_name, anchors=(limits[0], limits[1]))
 
+    def _set_default_ptype(self):
+        """Check if all parameters have a ptype that is in the list of allowed ptypes.
+
+        If no ptype is specified, set the default ptype "needs_reinit".
+
+        """
+        allowed_ptypes = ["rate", "shape", "efficiency", "livetime", "needs_reinit"]
+        default_ptype = "needs_reinit"
+        for p in self.parameters:
+            if p.ptype is None:
+                p.ptype = default_ptype
+            elif p.ptype not in allowed_ptypes:
+                raise ValueError(
+                    f"Parameter {p.name} has ptype {p.ptype} which is not in the list of "
+                    f"allowed ptypes: {allowed_ptypes}."
+                )
+
```

Comment:
I think you just need to set a default value at https://github.com/XENONnT/alea/blob/bd71e948d7a57191bd1fd28e750fe2e439c6f8c8/alea/parameters.py#L43 to `ptype`, and maybe also do any kind of test there.

If the `needs_reinit` is just for `BlueiceExtendedModel`, I think you should change the `parameter_definition ` at https://github.com/XENONnT/alea/pull/110/files#diff-ccb14fbc2f6819c7774476fe82d0c97563c7006391df64c86b0fbea0706af8f5L52 then. Because it is not a good practice to change the definition of `ptype` outside parameters.py.

Reply:
We don't want this to be the default. This means that you _always_ have to specify a ptype basically. The Gaussian Model for example would become more complex than it needs to be and this then also translates to any other likelihoods added in the future. I think it's ok to have functionality in parameters that you can take into your project if you like but don't have to.